extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTableAdapter.java

```diff
@@ -228,231 +229,164 @@ public synchronized Optional<Schema> schema(final int schemaId) {
     }
 
     /**
-     * Return {@link TableDefinition table definition}.
-     *
-     * @return The table definition
-     */
-    public TableDefinition definition() {
-        // Load the table from the catalog.
-        return definition(null);
-    }
-
-    /**
-     * Return {@link TableDefinition table definition} with optional instructions for customizations while reading.
-     *
-     * @param instructions The instructions for customizations while reading (or null for default instructions)
-     * @return The table definition
+     * Retrieves the appropriate {@link Snapshot} based on the provided {@link IcebergReadInstructions}, or {@code null}
+     * if no snapshot is provided.
      */
-    public TableDefinition definition(@Nullable final IcebergInstructions instructions) {
-        // Load the table from the catalog.
-        return definition(null, instructions);
+    @Nullable
+    private Snapshot getSnapshot(final IcebergReadInstructions readInstructions) {
+        final Snapshot snapshot;
+        if (readInstructions.snapshot().isPresent()) {
+            snapshot = readInstructions.snapshot().get();
+        } else if (readInstructions.tableSnapshotId().isPresent()) {
+            snapshot = snapshot(readInstructions.tableSnapshotId().getAsLong())
+                    .orElseThrow(() -> new IllegalArgumentException(
+                            "Snapshot with id " + readInstructions.tableSnapshotId().getAsLong() + " not found for " +
+                                    "table " + tableIdentifier));
+        } else {
+            snapshot = null;
+        }
+        return snapshot;
```

Comment:
I have no problem with this, but could rewrite to simply `return` the values rather than store.
```suggestion
        final Snapshot snapshot;
        if (readInstructions.snapshot().isPresent()) {
            return readInstructions.snapshot().get();
        }
        if (readInstructions.tableSnapshotId().isPresent()) {
            return snapshot(readInstructions.tableSnapshotId().getAsLong())
                    .orElseThrow(() -> new IllegalArgumentException(
                            "Snapshot with id " + readInstructions.tableSnapshotId().getAsLong() + " not found for " +
                                    "table " + tableIdentifier));
        return null;
```